inc/saihostintf.h

```diff
@@ -692,6 +587,179 @@ typedef sai_status_t(*sai_get_hostif_attribute_fn)(
         _Inout_ sai_attribute_t *attr_list);
 
 /**
+* @brief Attribute data for SAI_HOST_INTERFACE_TABLE_ENTRY_ATTR_TYPE
+*/
+typedef enum _sai_host_interface_table_entry_type_t
+{
+    /** Port-based Host Interface entry Type */
+    SAI_HOST_INTERFACE_TABLE_ENTRY_TYPE_PORT,
+
+    /** LAG based Host Interface entry Type */
+    SAI_HOST_INTERFACE_TABLE_ENTRY_TYPE_LAG,
+
+    /** Vlan based Host Interface entry Type */
+    SAI_HOST_INTERFACE_TABLE_ENTRY_TYPE_VLAN,
+
+    /** Wildcard Interface entry Type */
+    SAI_HOST_INTERFACE_TABLE_ENTRY_TYPE_TRAP_ID,
+
+    /** Wildcard Interface, wildcard trap id */
+    SAI_HOST_INTERFACE_TABLE_ENTRY_TYPE_WILDCARD
+
+} sai_host_interface_table_entry_type_t;
+
+/**
+* @brief Attribute data for SAI_HOST_INTERFACE_TABLE_ENTRY_ATTR_CHANNEL_TYPE
+*/
+typedef enum _sai_host_interface_table_entry_channel_type_t
+{
+    /** receive packets via callback */
+    SAI_HOST_INTERFACE_TABLE_ENTRY_CHANNEL_TYPE_CB,
+
+    /** receive packets via file descriptor */
+    SAI_HOST_INTERFACE_TABLE_ENTRY_CHANNEL_TYPE_FD,
+
+    /** receive packets via Linux netdev type port */
+    SAI_HOST_INTERFACE_TABLE_ENTRY_CHANNEL_TYPE_NETDEV_PHYSICAL_PORT,
+
+    /** receive packets via Linux netdev logical port (LAG or port) */
+    SAI_HOST_INTERFACE_TABLE_ENTRY_CHANNEL_TYPE_NETDEV_LOGICAL_PORT,
+
+    /** receive packets via Linux netdev L3 interface */
+    SAI_HOST_INTERFACE_TABLE_ENTRY_CHANNEL_TYPE_NETDEV_L3
+
+} sai_host_interface_table_entry_channel_type_t;
+
+/**
+* @brief Host interface table entry attribute IDs
```

Comment:
There is explanation of the host interface table entry and flows in proposal document v9 which is already committed in github
https://github.com/opencomputeproject/SAI/commit/8f277b732c4f0c369a56ce125c1304fff6400b1a